app/services/client_messaging_service.rb

```diff
@@ -1,71 +1,86 @@
+      replaced_body = ReplacementParametersService.new(**replacement_args).process
       OutgoingTextMessage.create!(
         client: client,
         to_phone_number: client.sms_phone_number,
         sent_at: DateTime.now,
         user: user,
-        body: body
+        body: replaced_body,
       )
     end
 
-    def send_system_text_message(client, body)
-      OutgoingTextMessage.create!(
-        client: client,
-        body: body,
-        to_phone_number: client.sms_phone_number,
-        sent_at: DateTime.now
-      )
+    def send_system_text_message(client:, body:, tax_return: nil, locale: nil)
+      args = { client: client, body: body, user: nil }
+      args[:tax_return] = tax_return if tax_return.present?
+      args[:locale] = locale if locale.present?
+      send_text_message(**args)
     end
 
-    def send_message_to_all_opted_in_contact_methods(client, user, body)
+    def send_message_to_all_opted_in_contact_methods(client:, user:, body:, locale: nil, tax_return: nil)
       message_records = {
         outgoing_email: nil,
         outgoing_text_message: nil,
       }
+      args = { client: client, user: user, body: body }
+      args[:tax_return] = tax_return if tax_return.present?
+      args[:locale] = locale if locale.present?
       if client.intake.email_notification_opt_in_yes? && client.email_address.present?
-        message_records[:outgoing_email] = send_email(client, user, body)
+        message_records[:outgoing_email] = send_email(**args)
       end
       if client.intake.sms_notification_opt_in_yes? && client.sms_phone_number.present?
-        message_records[:outgoing_text_message] = send_text_message(client, user, body)
+        message_records[:outgoing_text_message] = send_text_message(**args)
       end
       message_records
     end
 
-    def send_system_message_to_all_opted_in_contact_methods(client, email_body: nil, sms_body: nil, subject: nil)
+    def send_system_message_to_all_opted_in_contact_methods(client:, email_body: nil, sms_body: nil, subject: nil, tax_return: nil, locale: nil)
       message_records = {
         outgoing_email: nil,
         outgoing_text_message: nil,
       }
+      args = { client: client, body: email_body, subject: subject }
+      args[:tax_return] = tax_return if tax_return.present?
+      args[:locale] = locale if locale.present?
+
       if client.intake.email_notification_opt_in_yes? && client.email_address.present? && email_body.present?
-        message_records[:outgoing_email] = send_system_email(client, email_body, subject)
+        message_records[:outgoing_email] = send_system_email(**args)
       end
       if client.intake.sms_notification_opt_in_yes? && client.sms_phone_number.present? && sms_body.present?
-        message_records[:outgoing_text_message] = send_system_text_message(client, sms_body)
+        args.delete(:subject)
+        args[:body] = sms_body
+        message_records[:outgoing_text_message] = send_system_text_message(**args)
```

Comment:
I usually have a personal preference for seeing all arguments passed directly, rather than `**args`'d. Definitely not a blocker.